src/providers/eip-1559-gas-price-provider.ts

```diff
@@ -43,7 +43,11 @@ export class EIP1559GasPriceProvider extends IGasPriceProvider {
 
   public async getGasPrice(): Promise<GasPrice> {
     const feeHistoryRaw = (await this.provider.send('eth_feeHistory', [
-      this.blocksToConsider,
+      /**
+       * @note Use BigNumber.from(this.blocksToConsider).toHexString() after hardhat adds support
+       * @see https://github.com/NomicFoundation/hardhat/issues/1585 .___.
+       */
+      `0x${this.blocksToConsider}`,
```

Comment:
I think this would break if blocksToConsider is >= 10?? (0x10 = 16) ? A little confused as to why we have to do this, but I'm down ?

Reply:
Ah facepalm yeah that won't work

Going to opt for `BigNumber.from(this.blocksToConsider).toHexString().replace('0x0', '0x')` since it looks like the issue still exists: https://github.com/NomicFoundation/hardhat/issues/1585